javascript/javascript1/week4/class-exercises/sentiment-analyzer.js

```diff
@@ -0,0 +1,87 @@
+const positiveWordsList = [
+  'cool',
+  'awesome',
+  'super',
+  'amazing',
+  'fantastic',
+  'brilliant',
+  'remarkable',
+  'incredible',
+  'spectacular',
+  'outstanding',
+  'exceptional',
+  'wonderful',
+  'magnificent',
+  'phenomenal',
+  'happy',
+];
+
+const negativeWordsList = [
+  'boring',
+  'sad',
+  'upset',
+  'terrible',
+  'awful',
+  'horrible',
+  'dismal',
+  'dreary',
+  'gloomy',
+  'disappointing',
+  'lackluster',
+  'mediocre',
+  'troublesome',
+  'annoying',
+  'distressing',
+];
+
+function getCommonWordsArray(sentenceArray, wordsList) {
+  const commonWords = [];
+  for (let i = 0; i < sentenceArray.length; i++) {
+    const word = sentenceArray[i];
+    if (wordsList.includes(word)) {
+      commonWords.push(word);
+    }
+  }
+  return commonWords;
+}
+
+function getSentimentScore(string) {
+  const wordsFromString = string.split(' ');
+  const positiveWords = getCommonWordsArray(wordsFromString, positiveWordsList);
+  const negativeWords = getCommonWordsArray(wordsFromString, negativeWordsList);
+  const score = positiveWords.length - negativeWords.length;
+  return { score, positiveWords, negativeWords };
+}
+
+const sentimentScoreObject = getSentimentScore('I am mega super awesome happy');
```

Comment:
Grammatically, those would have to be comma-separated.
```suggestion
const sentimentScoreObject = getSentimentScore('I am mega, super, awesome, happy!');
```